bundle/edu.gemini.itc.web/src/main/java/edu/gemini/itc/web/html/PlannedTimeMath.java

```diff
@@ -0,0 +1,120 @@
+package edu.gemini.itc.web.html;
+
+import edu.gemini.shared.util.immutable.None;
+import edu.gemini.shared.util.immutable.Option;
+import edu.gemini.shared.util.immutable.Some;
+import edu.gemini.spModel.config2.Config;
+import edu.gemini.spModel.config2.ConfigSequence;
+import edu.gemini.spModel.config2.ItemKey;
+import edu.gemini.spModel.guide.StandardGuideOptions;
+import edu.gemini.spModel.obs.plannedtime.PlannedTime;
+import edu.gemini.spModel.obs.plannedtime.PlannedTimeCalculator;
+import edu.gemini.spModel.obs.plannedtime.SetupTime;
+import edu.gemini.spModel.obscomp.ItcOverheadProvider;
+import edu.gemini.spModel.time.ChargeClass;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+// Extracted from edu.gemini.spModel.obs.plannedtime.PlannedTime since it
+// doesn't seem useful elsewhere and indeed is only used by this package.
+
+public final class PlannedTimeMath {
```

Comment:
As noted, I just moved this code out of `PlannedTime` with only minimal changes.  I considered making [an updated version](https://gist.github.com/swalker2m/206b7563b2719b508837f5332d98890a) but decided against it to avoid potentially introducing bugs.